src/common/scaled-icon-buffer.c

```diff
@@ -12,6 +14,26 @@
 #include "desktop-entry.h"
 #include "img/img.h"
 #include "node.h"
+#include "view.h"
+
+#if HAVE_LIBSFDO
+static struct lab_data_buffer *
+choose_best_buffer(struct scaled_icon_buffer *self, int icon_size, double scale)
+{
+	int min_dist = INT_MAX;
+	struct lab_data_buffer *best_buffer = NULL;
+
+	struct lab_data_buffer **buffer;
+	wl_array_for_each(buffer, &self->view->icon.buffers) {
+		int dist = abs((*buffer)->base.width - (int)(icon_size * scale));
+		if (dist < min_dist) {
```

Comment:
I think we should prefer using matching or bigger icons here and then downscale rather than upscale smaller ones, even if they are closer to our desired icon size.